lint/ruff.bzl

```diff
@@ -186,6 +190,7 @@ ruff_workaround_20269 = repository_rule(
     attrs = {
         "build_file_content": attr.string(),
         "sha256": attr.string(),
+        "strip_prefix": attr.string(doc = "unlike http_archive, any value causes us to pass --strip-components=1 to tar"),
```

Comment:
correct, this rule has to be duck-typed to look like http_archive since we swap it out below.

I would spend more effort to make this nice, but the underlying problem was fixed in Bazel 7.2.0 so the workaround can go away in a year or two.